src/algorithms/machinelearning/tensorflowpredictfsdsinet.cpp

```diff
@@ -0,0 +1,251 @@
+  int patchHopSize = parameter("patchHopSize").toInt();
+  int batchSize = parameter("batchSize").toInt();
+
+  int frameSize = 660;
+  int hopSize = 220;
+  int numberBands = 96;
+  int patchSize = 101;
+
+  vector<int> inputShape({batchSize, 1, patchSize, numberBands});
+
+  _frameCutter->configure("frameSize", frameSize, "hopSize", hopSize);
+
+  _vectorRealToTensor->configure("shape", inputShape,
+                                 "lastPatchMode", lastPatchMode,
+                                 "patchHopSize", patchHopSize);
+
+  _configured = true;
+
+  string input = parameter("input").toString();
+  string output = parameter("output").toString();
+
+  // {Batch, Channel, Time, Freq} --> {Batch, Time, Freq, Channel}
+  vector<int> permutation({0, 2, 3, 1});
+  _tensorTranspose->configure("permutation", permutation);
+  _tensorToPool->configure("namespace", input);
+  _poolToTensor->configure("namespace", output);
+
+  string graphFilename = parameter("graphFilename").toString();
+  string savedModel = parameter("savedModel").toString();
+
+  _tensorflowPredict->configure("graphFilename", graphFilename,
+                                "savedModel", savedModel,
+                                "inputs", vector<string>({input}),
+                                "outputs", vector<string>({output}),
+                                "squeeze", false,
+                                "isTrainingName", "");
+}
+
+} // namespace streaming
+} // namespace essentia
+
+
+
+namespace essentia {
+namespace standard {
+
+const char* TensorflowPredictFSDSINet::name = "TensorflowPredictFSDSINet";
+const char* TensorflowPredictFSDSINet::category = "Machine Learning";
+const char* TensorflowPredictFSDSINet::description = DOC(
+  "This algorithm makes predictions using FSD-SINet models.\n"
+  "\n"
+  "Internally, it uses TensorflowInputFSDSINet for the input feature extraction "
+  "(mel bands). It feeds the model with patches of 101 mel-band frames and "
+  "jumps a constant amount of frames determined by `patchHopSize`.\n"
+  "\n"
+  "By setting the `batchSize` parameter to -1 or 0 the patches are stored to run a single "
+  "TensorFlow session at the end of the stream. This allows to take advantage "
+  "of parallelization when GPUs are available, but at the same time it can be "
+  "memory exhausting for long files.\n"
+  "\n"
+  "The recommended pipeline is as follows::\n"
+  "\n"
+  "  MonoLoader(sampleRate=22050) >> TensorflowPredictFSDSINet\n"
+  "\n"
+  "Note: This algorithm does not make any check on the input model so it is "
+  "the user's responsibility to make sure it is a valid one.\n"
+  "\n"
+  "Note II: The FSD-SINet models were trained on normalized audio clips. "
+  "Clip-level normalization is only implemented in standard mode since in streaming there is no access to the entire audio clip. "
+  "In this case, it is the responsibility of the user to feed the algorithm with audio featuring a healthy dynamic range.\n\n"
```

Comment:
It is not clear what is the "healthy" range that is expected. Can it be more concrete?